sklearn/utils/tests/test_extmath.py

```diff
@@ -161,6 +162,85 @@ def test_randomized_svd_low_rank_all_dtypes(dtype):
     check_randomized_svd_low_rank(dtype)
 
 
+@pytest.mark.parametrize('dtype',
+                         (np.int32, np.int64, np.float32, np.float64))
+def test_randomized_eigsh(dtype):
+    """Test that `_randomized_eigsh` returns the appropriate components"""
+
+    X = np.diag(np.array([1., -2., 0., 3.], dtype=dtype))
+
+    # with 'module' selection method, the negative eigenvalue shows up
+    lambd_, alph_ = _randomized_eigsh(X, n_components=2, selection='module')
+    # eigenvalues
+    assert lambd_.shape == (2,)
+    assert_array_almost_equal(lambd_, [3., -2.])  # negative eigenvalue here
+    # eigenvectors
+    assert alph_.shape == (4, 2)
```

Comment:
Wouldn't it make more sense to return the transpose with shape `(n_components, n_samples)` instead?

Reply:
Scratch that, let's follow the convention of `scipy.linalg.eigh`.

However why the name alpha? Wouldn't it make more sence to call that `eigen_vectors` more explicitly?

We could also rename the lambdas to `eigen_values`.

Reply:
Oh I totally agree with this comment. I do not like this naming at all, it is confusing. Unfortunately it is the one used in KernelPCA, and here I did not realize that I had the freedom to choose a better one since this is a test :)
I'll change it happily now

Reply:
I eventually used `eigvals, eigvecs`